ev3dev/core.py

```diff
@@ -146,7 +143,7 @@ def __init__(self, class_name, name_pattern='*', name_exact=False, **kwargs):
         def get_index(file):
             match = Device._DEVICE_INDEX.match(file)
             if match:
-                return int(match.group('idx'))
```

Comment:
MicroPython's implementation of the `re` module is barebones and doesn't have support for named groups. I don't see any reason why we need to use a named group anyway.